unmarshal.go

```diff
@@ -137,5 +152,38 @@ func readElement(r0 io.Reader, n int64, vo reflect.Value) (io.Reader, error) {
 				}
 			}
 		}
+		pos += headerSize + size
+	}
+}
+
+type UnmarshalHook func(elem *Element)
```

Comment:
Functional options implementation which I intended is
```go
type UnmarshalOptions struct {
	hooks []func(*Element)
}

type UnmarshalOption func(*UnmarshalOptions)

func WithElementHooks(hooks ...func(*Element)) UnmarshalOption {
	return func(opts *UnmarshalOptions) {
		opts.hooks = hooks
	}
}

func Unmarshal(r io.Reader, val interface{}, opts ...UnmarshalOption) error {
```
which allows to add another non-hook options without changing API.

Reply:
Thank you for explanation. I will update so 😄

Reply:
Updated 8517844982554139f20676f38f6927a7d4689611